lib/awesome_spawn.rb

```diff
@@ -62,6 +62,7 @@ def logger
   # @option options [Hash,Array] :params The command line parameters. See
   #   {#build_command_line} for how to specify params.
   # @option options [String] :in_data Data to be passed on stdin.
+  # @option options [Boolean] :capture2e Combine STDOUT/STDERR streams from run command
```

Comment:
Not sure about a dedicated option, or if we should just allow `[:out, :err]` as per Kernel.spawn.  If we do keep the option, I'm not sure capture2e is a good name (you'd have to know and understand open3 terminology, which I think is weird).  I don't really have an alternate suggestion though.

Reply:
Yeah, I was struggling a bit on this one as well.  Some reasons "why" I did, but aren't necessarily reasons we "should":

- `:capture2e` was not taken as a reserved key by this lib or in Ruby's `.spawn` or `Open3`
- Doubtful it would be used as a key by those libs in the future
- While does require knowing `Open3` terminology, however, once one does I think it is pretty obvious what it is doing (I am of the opinion that I don't think it hurts that others learn about the `Open3` lib in Ruby)
- A key like this makes it "patch release" worthy versus "major version release" worthy.

Again, not all "great" reasons, just some of them.

Reply:
on the name, I'm thinking `:capture_error => true` reads better, personally.

Reply:
I don't know, I feel like that means "capture the error, because otherwise we drop it on the floor", which isn't true.

Maybe:

- `:combined_output => true`?
- `:shared_output => true`?
- `:outerr => true` ?
- `:super_sekret_special_sauce => "yes_please!!1!"`?

I might not be serious about one of them...

Reply:
I was thinking `:combined_output => true` or `:redirect_stderr` before I got to the last comment.

Reply:
@bdunne does this mean you would prefer one of the ones you are suggesting, or just bringing up some more alternatives to the conversation?

I am fine with `combined_output` if you would rather use that, I just think that `:redirect_stderr` is ambiguous as it could imply `2>/dev/null` as well.

Reply:
> I just think that `:redirect_stderr` is ambiguous as it could imply `2>/dev/null` as well.

Good point.

Reply:
So far my favorite is `combined_output`.

> I don't know, I feel like that means "capture the error, because otherwise we drop it on the floor", which isn't true isn't the case without this option.

Agreed.  The name should imply that the outputs are being combined or merged.

Reply:
I can switch it to `:combined_output`.  Not a huge change to do.

Reply:
Okay, went with `:combined_output` and rebased out the `[FIXUP]` commit for it.  Resolving this thread.